src/cmake/modules/FindOpenEXR.cmake

```diff
@@ -71,16 +71,28 @@ endif ()
 
 # List of likely places to find the libraries -- note priority override of
 # OPENEXR_CUSTOM_LIB_DIR and ${OPENEXR_HOME}/lib.
-set (GENERIC_LIBRARY_PATHS 
-    ${OPENEXR_CUSTOM_LIB_DIR}
-    ${OPENEXR_HOME}/lib
-    ${ILMBASE_HOME}/lib
+
+# If there's no OPENEXR_HOME or ILMBASE_HOME, then the path will point to
+# "/lib", which may not always be wanted/expected.
+if (${OPENEXR_CUSTOM_LIB_DIR})
+  set (GENERIC_LIBRARY_PATHS ${OPENEXR_CUSTOM_LIB_DIR})
+endif()
+
+if (${OPENEXR_HOME})
+  set (GENERIC_LIBRARY_PATHS ${OPENEXR_HOME})
+endif()
+
+if (${ILMBASE_HOME})
+  set (GENERIC_LIBRARY_PATHS ${ILMBASE_HOME})
+endif()
+
+set (GENERIC_LIBRARY_PATHS
```

Comment:
Won't this lose the GENERIC_LIBRARY_PATHS value that you just set?
Maybe these should be successively appending rather than assigning?